tools/FindPipelinesUsingRetiredImages/QueryJobHistoryForRetiredImages.ps1

```diff
@@ -0,0 +1,102 @@
+param (
+    $accountUrl,
+    $pat
+)
+
+# Create the VSTS auth header
+$base64authinfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":$pat"))
+$vstsAuthHeader = @{"Authorization"="Basic $base64authinfo"}
+$allHeaders = $vstsAuthHeader + @{"Content-Type"="application/json"; "Accept"="application/json"}
+
+try
+{
+	$result = Invoke-WebRequest -Headers $allHeaders -Method GET "$accountUrl/_apis/DistributedTask/pools?api-version=5.0-preview"
+	if ($result.StatusCode -ne 200)
+    {
+		echo $result.Content
+		throw "Failed to query pools"
+	}
+	$resultJson = ConvertFrom-Json $result.Content
+    $azurePipelinesPoolId = 0
+    foreach($pool in $resultJson.value)
+    {
+        if ($pool.name -eq "Azure Pipelines")
+        {
+            $azurePipelinesPoolId = $pool.id
+            break
+        }
+    }
+
+    if ($azurePipelinesPoolId -eq 0)
+    {
+        throw "Failed to find Azure Pipelines pool"
+    }
+    
+    Write-Host ("Azure Pipelines Pool Id: " + $azurePipelinesPoolId)
+
+    $msg = 'Query next 200 jobs? (y/n)'
```

Comment:
would we not expect customers to want to query some specified number or all of them?

Reply:
I assume that we want to query all of them. The REST API is limited to 200 at a time. I supposed I could take input for how many to query, and just run it until we get that many back?

Let me think on it. I don't know how much more time I want to spend on this script.

Reply:
Yeah I am going to leave this alone for now